### DISCUSSION TITLE
Cake for .NET 4.5.1

### DISCUSSION THREAD

**Discussion by Aynur11** [cake-build/cake#3359] [Extension Q&A] [answered]
Hello everyone. Are there any way to install the Cake package into a project that targets .NET Version 4.5.1?
When I try this I get this: "Could not install package 'Cake'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.5.1', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author."
The way to up .NET version is not acceptable for my project.

**Comment by augustoproiete** [comment, marked as answer]
Hi @Aynur11, are you trying to use [Cake Frosting](https://cakebuild.net/docs/running-builds/runners/cake-frosting) or [Cake Scripting](https://cakebuild.net/docs/running-builds/runners/dotnet-tool) (i.e. `.cake` files)?

If you are using Cake Frosting, then the answer is **no**. The [minimum .NET Framework version](https://github.com/cake-build/cake/blob/v1.1.0/src/Cake.Frosting/Cake.Frosting.csproj#L4) to install the latest version of Cake Frosting as of this writing is .NET Framework 4.6.1 (or .NET Core 2.1/3.1 or .NET 5).

If you are using Cake Scripting, then you don't need to _install_ Cake packages on your project, and the Cake runner is external, which gives you the option to use .NET Core 2.1/3.1 or .NET 5, for example. As long as you install a compatible .NET Core version on the machine, you will be able to run the `dotnet-cake` runner and build your .NET 4.6.1 project (from a `.cake` script).

---

[![image](https://user-images.githubusercontent.com/177608/118989996-e4187200-b958-11eb-8bbf-dd784a035926.png)](https://cakebuild.net/docs/running-builds/runners/)
https://cakebuild.net/docs/running-builds/runners/

**Reply by devlead** [reply]
That said you can still build something that targets v4.5.1, it's just that Cake needs a newer framework to execute and orchestrate that build.

**Reply by Aynur11** [reply]
> Hi @Aynur11, are you trying to use [Cake Frosting](https://cakebuild.net/docs/running-builds/runners/cake-frosting) or [Cake Scripting](https://cakebuild.net/docs/running-builds/runners/dotnet-tool) (i.e. `.cake` files)?
> 
> If you are using Cake Frosting, then the answer is **no**. The [minimum .NET Framework version](https://github.com/cake-build/cake/blob/v1.1.0/src/Cake.Frosting/Cake.Frosting.csproj#L4) to install the latest version of Cake Frosting as of this writing is .NET Framework 4.6.1 (or .NET Core 2.1/3.1 or .NET 5).
> 
> If you are using Cake Scripting, then you don't need to _install_ Cake packages on your project, and the Cake runner is external, which gives you the option to use .NET Core 2.1/3.1 or .NET 5, for example. As long as you install a compatible .NET Core version on the machine, you will be able to run the `dotnet-cake` runner and build your .NET 4.6.1 project (from a `.cake` script).
> 
> [![image](https://user-images.githubusercontent.com/177608/118989996-e4187200-b958-11eb-8bbf-dd784a035926.png)](https://cakebuild.net/docs/running-builds/runners/)
> https://cakebuild.net/docs/running-builds/runners/

Probably I am using Cake Scripting.
I just have found https://github.com/cake-build/resources/blob/master/dotnet-framework/build.ps1
This build.ps1 will correctly interract with ,cake file to build my project?
I just dont want to include dotnet commands because CI server might work only with .NET Framework (without .NET Core).

**Reply by augustoproiete** [reply]
Yes, the Powershell bootstrapper that you linked will call the `Cake.exe` runner.

You will need to have at least .NET 4.6.1 installed on the CI server for `Cake.exe` to run, but your project (e.g. `.csproj`) can still target .NET 4.5.1 without issues.

There's no relationship between the .NET version that Cake needs to run, and the .NET version that your project needs to be compiled against. Behind the scenes, Cake will just run MSBuild to compile your project.

**Reply by Aynur11** [reply]
Thank you so much!
